tsdb/exemplar.go

```diff
@@ -305,13 +310,14 @@ func (ce *CircularExemplarStorage) Resize(l int64) int {
 // migrate is like AddExemplar but reuses existing structs. Expected to be called in batch and requires
 // external lock and does not compute metrics.
 func (ce *CircularExemplarStorage) migrate(entry *circularBufferEntry) {
-	seriesLabels := entry.ref.seriesLabels.String()
+	var buf [1024]byte
+	seriesLabels := entry.ref.seriesLabels.Bytes(buf[:])
```

Comment:
Since this is a repeated pattern, it would be more intuitive and handy to have a function like this which can be used everywhere
```go
func getIndexKey(l labels.Labels) []byte {
	var buf [1024]byte
	return l.Bytes(buf[:])
}
```

Reply:
Does that avoid the allocation?  I guess it might, if the whole function gets inlined, but that is unpredictable.

Reply:
Not sure about that